### PR TITLE
ci(e2e): implement forkproxy

### PR DESCRIPTION
**Pull request by corverroos**
Add the forkproxy binary (and docker image already manually upload to dockerhub).

Forkproxy:
- It wraps anvil an internal anvil instance
- Proxying requests to that instance

TODO:
- Add support for forking the anvil instance if configured.
- Proxy requests to forked instance. 

task: none

### CODE REVIEW

File: e2e/docker/compose.yaml.tmpl [hunk 1]
```diff
@@ -34,20 +34,15 @@ services:
 {{end}}
 
 {{- range .Anvils }}
-  # Initialises geth files and folder from provided genesis file.
   {{ .Chain.Name }}:
     labels:
       e2e: true
     container_name: {{ .Chain.Name }}
     platform: linux/amd64
```
Comment (pavelnikolov) [inline e2e/docker/compose.yaml.tmpl:41]: Do we still need the platform here?

**Review by pavelnikolov — APPROVED** [review]

File: e2e/docker/compose.yaml.tmpl (hunk 1 above)
Comment (corverroos) [reply e2e/docker/compose.yaml.tmpl:41]: I'm not sure

File: e2e/forkproxy/app/app.go [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+	"github.com/omni-network/omni/lib/log"
+
+	"golang.org/x/sync/errgroup"
+)
+
+type Config struct {
+	ListenAddr    string
+	ChainID       uint64
+	LoadState     string
+	BlockTimeSecs uint64
```
Comment (pavelnikolov) [inline e2e/forkproxy/app/app.go:28]: Does it make sense for this to be `time.Duration`?

Comment (corverroos) [reply e2e/forkproxy/app/app.go:28]: I'm sticking to the anvil API for consistency

**Review by kevinhalliday — APPROVED** [review]

File: e2e/docker/compose.yaml.tmpl (hunk 1 above)
Comment (kevinhalliday) [reply e2e/docker/compose.yaml.tmpl:41]: We've only build for [amd64](https://hub.docker.com/repository/docker/omniops/forkproxy/tags) so yes

**Pull request merged**